src/Backend/Modules/Pages/Actions/Settings.php

```diff
@@ -22,32 +33,95 @@ public function execute(): void
-            'meta_navigation',
-            $this->get('fork.settings')->get($this->getModule(), 'meta_navigation', false)
+        $this->settingsForm = $this->createForm(
+            SettingsType::class,
+            new SettingsDataTransferObject(
+                $this->get('fork.settings'),
+                $this->get('media_library.repository.group')
+            )
         );
+
+        $this->template->assign('form', $this->settingsForm->createView());
     }
 
     private function validateForm(): void
     {
-        // form is submitted
-        if ($this->form->isSubmitted()) {
-            // form is validated
-            if ($this->form->isCorrect()) {
-                // set our settings
+        $this->settingsForm->handleRequest($this->getRequest());
+
+        // settingsForm is submitted
+        if ($this->settingsForm->isSubmitted() && $this->settingsForm->isValid()) {
+            /** @var SettingsDataTransferObject $data */
+            $data = $this->settingsForm->getData();
+
+            // set our settings
+            $this->get('fork.settings')->set(
+                $this->getModule(),
+                'meta_navigation',
+                $data->metaNavigation
+            );
+
+            $this->get('fork.settings')->set(
+                $this->getModule(),
+                'offline_title_' . Language::getWorkingLanguage(),
+                $data->offlineTitle
+            );
+
+            $this->get('fork.settings')->set(
+                $this->getModule(),
+                'offline_text_' . Language::getWorkingLanguage(),
+                $data->offlineText
+            );
+
+            if ($data->offlineImage instanceof MediaGroup) {
+                $this->get('media_library.repository.group')->add($data->offlineImage);
+                $this->get('doctrine.orm.entity_manager')->flush();
                 $this->get('fork.settings')->set(
                     $this->getModule(),
-                    'meta_navigation',
-                    (bool) $this->form->getField('meta_navigation')->getValue()
+                    'offline_image_' . Language::getWorkingLanguage(),
+                    $data->offlineImage->getId()
                 );
-
-                // redirect to the settings page
-                $this->redirect(BackendModel::createUrlForAction('Settings') . '&report=saved');
             }
+
+            $this->renderOfflinePage();
+
+            // redirect to the settings page
+            $this->redirect(BackendModel::createUrlForAction('Settings') . '&report=saved');
         }
     }
+
+    private function renderOfflinePage(): void
+    {
+        /** @var Environment $twig */
+        $twig = $this->get('twig');
+
+        /** @var ParseBlocksExtension $extension */
+        $extension = $twig->getExtension(ParseBlocksExtension::class);
+
+        $html = $twig->render(
+            BACKEND_MODULES_PATH . '/' . $this->getModule() .'/Layout/Templates/Offline.html.twig',
```

Comment:
wouldn't it make more sense to put this in the frontend? That way you can also override it in your theme